### PR TITLE
Update sphinx-scylladb-theme to 1.7.1

### LINKED ISSUE #1893: Update to use latest sphinx-scylladb-theme theme
Latest sphinx-scylladb-theme contains several bug fixes we asked for so we should bump to >=1.7.1. This should make our docs look quite better.

### PR DESCRIPTION
**Pull request by rzetelskik**
**Description of your changes:** Update sphinx-scylladb-theme docs dependency to 1.7.1. Most notably, it adds https://github.com/scylladb/sphinx-scylladb-theme/pull/1051. 

**Which issue is resolved by this Pull Request:**
Resolves #1893

/kind machinery
/priority important-soon

### CODE REVIEW

File: docs/pyproject.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ authors = ["ScyllaDB Documentation Contributors"]
 python = "^3.9"
 pyyaml = "6.0.1"
 pygments = "2.15.1"
-sphinx-scylladb-theme = "~1.6.6"
+sphinx-scylladb-theme = "^1.7.1"
```
Comment (tnozicka) [inline docs/pyproject.toml:12]:
```suggestion
sphinx-scylladb-theme = "~1.7.1"
```
`minor` upgrades need manual changes

**Review by tnozicka — CHANGES_REQUESTED** [review]

File: docs/pyproject.toml (hunk 1 above)
Comment (rzetelskik) [reply docs/pyproject.toml]: done

**Comment by tnozicka** [comment]
hm, the menu is gone
/hold

Screenshot from 2024-04-16 12-00-05

**Comment by rzetelskik** [comment]
@tnozicka works for me, including the auto scroll
image

**Comment by tnozicka** [comment]
I've run `poetry install && make preview` inside the container. That used to work before, I can try this on a clean system if you confirm it works for you but there can be something hidden related to the upgrade.

**Comment by tnozicka** [comment]
> make setup dirhtml multiversionpreview

btw. those are run in parallel / racy (modulo explicit deps order)

**Comment by rzetelskik** [comment]
`podman run -it --pull=Always --rm -v="$( pwd )/docs:/go/$( go list -m )/docs:Z" --workdir="/go/$( go list -m )/docs" -p 5500:5500 quay.io/scylladb/scylla-operator-images:poetry-1.5 bash -euExo pipefail -O inherit_errexit -c 'poetry install && make preview'` 

This also works fine for me.

image

Did you clean the build dir before running it?

**Comment by tnozicka** [comment]
never had to but it helped
/shrug

/approve
/lgtm

**Comment by rzetelskik** [comment]
@tnozicka can we cancel the hold then?

**Pull request merged**